multiversx/mx-chain-go
# scr ordering on generation/execution

## Reasoning behind the pull request
The execution order of SCRs is currently estimated (post processing) rather than taken from execution, which makes it less efficient and in some cases is more prone to errors.

This PR will make the SCRs ordering available from the execution component directly and integrate the ordering into the outport driver.
  
## Proposed changes
On generation (execution output results), the smart contract results should be added to the ordering component, which should be made available on the post processing component as well as on the processors and preprocessors (already done in the feat branch).

The ordering component will give the correct relative order to the generated SCRs upon insertion.
If the SCRs are intra shard, then the given order means generation order as well as execution order.

## Testing procedure
* Run the execution order validation scripts for testing
* Check indexed blocks, for every type of transaction - regular move balance tx, relayed, builtin function call, rewards tx, smart contract result, contract call, for both successful and failed txs and check that the ordering is correct.

## Pre-requisites
- adapt the execution order validation scripts
- enable the debug component for the execution order during testing 

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?
